pkg/sdk/go/errors/errors.go

```diff
@@ -0,0 +1,182 @@
+// Copyright (c) Mainflux
+// SPDX-License-Identifier: Apache-2.0
+
+package errors
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+)
+
+var (
+	errFailedToReadBody = New("failed to read http response body ")
+	errRespBodyNotJSON  = New("response body is not vaild JSON ")
+	errJSONKeyNotFound  = New("response body expected error message json key not found ")
+)
+
+// Error type for SDK
+type Error interface {
+	// Error implements the error interface.
+	Error() string
+
+	// Msg returns error message
+	Msg() string
+
+	// Err returns wrapped error
+	Err() Error
+
+	// StatusCode returns error status code
+	StatusCode() int
+
+	// Status returns error Status
+	Status() string
+}
+
+var _ Error = (*customError)(nil)
+
+// customError represents a Mainflux  SDK error
+type customError struct {
+	statusCode int
+	status     string
```

Comment:
Do we need both code and status (since it's fixed relation between status and code)?

Reply:
No need , removed status